### PR TITLE
Improve integration test harness

### PR DESCRIPTION
**Pull request by kitsonk**
This PR improves the flexibility of the integration tests, making it easier to test various command line arguments and configurations as part of the integration test suite.  It also decouples the need to ensure the test input program and output text are no longer coupled together, can be re-used by multiple tests, and exit codes are no longer specified as part of the filename.

Tests are now specified with a `.test` file that is a simple configuration format where each option is specified on a single line. The key is the string to the left of the `:` deliminator and the value is the string to the right.

The keys currently supported are:

| Key         | Required | Description                                                                                                                                                               |
| ----------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `args` | Yes      | Specifies the command line arguments for the test. This should typically be input script for the test and a `--reload` to help ensure Deno doesn't leverage the cache.    |
| `output`    | Yes      | This is a text file which represents the output of the command. The string `[WILDCARD]` can be used in the output to specify ranges of text which any output is accepted. |
| `exit_code` | No       | If not present, it is assumed the script would exit normally (`0`). If specified, the harness will ensure the proper code is received.                                    |

And output looks something like this:

```
Running: Throw non-error...
  Command: /Users/kkelly/github/deno/out/debug/deno tests/error_007_any.ts --reload
Compiling /Users/kkelly/github/deno/tests/error_007_any.ts
  ✅  PASS

Running: CheckJS type error...
  Command: /Users/kkelly/github/deno/out/debug/deno tests/error_008_checkjs.js --reload
Compiling /Users/kkelly/github/deno/tests/error_008_checkjs.js
  ✅  PASS

Running: Exit with code 42...
  Command: /Users/kkelly/github/deno/out/debug/deno tests/exit_error42.ts --reload
Compiling /Users/kkelly/github/deno/tests/exit_error42.ts
  ✅  PASS

Running: Import HTTPS module...
  Command: /Users/kkelly/github/deno/out/debug/deno tests/https_import.ts --reload
Compiling /Users/kkelly/github/deno/tests/https_import.ts
Compiling https://gist.githubusercontent.com/ry/f12b2aa3409e6b52645bc346a9e22929/raw/79318f239f51d764384a8bded8d7c6a833610dde/print_hello.ts
  ✅  PASS

Running: --types flag...
  Command: /Users/kkelly/github/deno/out/debug/deno --types
  ✅  PASS
```

### CODE REVIEW

File: tests/012_async.test [hunk 1]
```diff
@@ -0,0 +1,3 @@
+title: async...await
```
Comment (ry) [inline tests/012_async.test]: Let's leave out the title. The file name is good enough. 

1) It's more minimal
2) It opens up the possibility that only the title is displayed - which take greping to associate with the file ... the filename immediately resolves

File: tests/019_media_types.test [hunk 2]
```diff
@@ -0,0 +1,3 @@
+title: Media types respected when loading modules
+deno_args: tests/019_media_types.ts --reload
```
Comment (ry) [inline tests/019_media_types.test]: rename deno_args to args

File: tools/integration_tests.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            sys.exit(1)
+
+        if pattern_match(expected_out, actual_out) != True:
+            print "  🚫  FAIL"
+            print "Expected output does not match actual."
+            print "Expected output: \n" + expected_out
+            print "Actual output:   \n" + actual_out
+            sys.exit(1)
+
+        print "  ✅  PASS"
```
Comment (ry) [inline tools/integration_tests.py]: Use the green "ok" that we use in the unit tests.

File: tools/integration_tests.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+        if exit_code != actual_code:
+            print "  🚫  FAIL"
+            print "Expected exit code %d but got %d" % (exit_code, actual_code)
+            print "Output:"
+            print actual_out
+            sys.exit(1)
+
+        if pattern_match(expected_out, actual_out) != True:
+            print "  🚫  FAIL"
```
Comment (ry) [inline tools/integration_tests.py]: Use the red fail we use in the unit tests. 
https://github.com/denoland/deno/blob/edd12b90a8cc169bb226fcaf273973765844ce0d/js/testing/testing.ts#L61-L71

**Comment by kitsonk** [comment]
Feedback addressed.  Output now looks like this:

```
test 001_hello.test
/Users/kkelly/github/deno/out/debug/deno tests/001_hello.js --reload
Compiling /Users/kkelly/github/deno/tests/001_hello.js
... ok
test 002_hello.test
/Users/kkelly/github/deno/out/debug/deno tests/002_hello.ts --reload
Compiling /Users/kkelly/github/deno/tests/002_hello.ts
... ok
test 003_relative_import.test
/Users/kkelly/github/deno/out/debug/deno tests/003_relative_import.ts --reload
Compiling /Users/kkelly/github/deno/tests/003_relative_import.ts
Compiling /Users/kkelly/github/deno/tests/subdir/print_hello.ts
... ok
```

**Pull request merged**